### PR TITLE
[fix][doc] Clarify order processing semantics for Key Shared Subscription docs

### PR DESCRIPTION
**Pull request by asafm**
### Modifications

* Clarify intention around processing semantics for Key Shared Subscription documentation
* Removed IntelliJ errors on markdown formatting for the file edited

### Verifying this change

- [ ] Make sure that the change passes the CI checks.

### Does this pull request potentially affect one of the following parts:

*If the box was checked, please highlight the changes*

- [ ] Dependencies (add or upgrade a dependency)
- [ ] The public API
- [ ] The schema
- [ ] The default values of configurations
- [ ] The threading model
- [ ] The binary protocol
- [ ] The REST endpoints
- [ ] The admin CLI options
- [ ] The metrics
- [ ] Anything that affects deployment

### Documentation

- [x] `doc` 
- [ ] `doc-required` 
- [ ] `doc-not-needed` 
- [ ] `doc-complete`

### CODE REVIEW

File: site2/docs/concepts-messaging.md [hunk 1]
```diff
@@ -743,7 +743,7 @@ Consistent Hashing will be used instead of Hash Range for Auto-split if the brok
...
+Key Shared Subscription type guarantees a key will be processed by a *single* consumer at any given time. When a new consumer is connected, some keys will change their mapping from existing consumers to the new consumer. Once the connection has been established, the broker will record the current read position and associate it with the new consumer. The read position is a marker indicating that messages have been dispatched to the consumers up to this point, and after it, no messages have been dispatched yet. The broker will start delivering messages to the new consumer *only* when all messages up to the read position have been acknowledged. This will guarantee that a certain key is processed by a single consumer at any given time. The trade-off is that if one of the existing consumers is stuck and no time-out was defined (acknowledging for you), the new consumer won't receive any messages until the stuck consumer resumes or gets disconnected.
```
Comment (asafm) [inline site2/docs/concepts-messaging.md:746]: image

File: site2/docs/concepts-messaging.md [hunk 2]
```diff
@@ -61,9 +61,9 @@ A producer is a process that attaches to a topic and publishes messages to a Pul
 
 Producers send messages to brokers synchronously (sync) or asynchronously (async).
 
-| Mode       | Description |
-|:-----------|-----------|
-| Sync send  | The producer waits for an acknowledgment from the broker after sending every message. If the acknowledgment is not received, the producer treats the sending operation as a failure.                                                                                                                                                                                    |
+| Mode       | Description                                                                                                                                                                                                                                                                                                                                                      |
```
Comment (asafm) [inline site2/docs/concepts-messaging.md:64]: image

File: site2/docs/concepts-messaging.md [hunk 3]
```diff
@@ -280,16 +280,16 @@ Consumer<byte[]> consumer = pulsarClient.newConsumer()
...
-:--------------------|:-----------
-1 | 1 seconds
-2 | 2 seconds
-3 | 4 seconds
-4 | 8 seconds
-5 | 16 seconds
-6 | 32 seconds
-7 | 60 seconds
-8 | 60 seconds
+| Redelivery count | Redelivery delay |
```
Comment (asafm) [inline site2/docs/concepts-messaging.md:283]: image

File: site2/docs/concepts-messaging.md [hunk 4]
```diff
@@ -324,16 +324,16 @@ consumer.ackTimeout(10, TimeUnit.SECOND)
...
-:--------------------|:-----------
-1 | 10 + 1 seconds
-2 | 10 + 2 seconds
-3 | 10 + 4 seconds
-4 | 10 + 8 seconds
-5 | 10 + 16 seconds
-6 | 10 + 32 seconds
-7 | 10 + 60 seconds
-8 | 10 + 60 seconds
+| Redelivery count | Redelivery delay |
```
Comment (asafm) [inline site2/docs/concepts-messaging.md:327]: image

File: site2/docs/concepts-messaging.md [hunk 5]
```diff
@@ -408,12 +408,12 @@ Consumer<byte[]> consumer = pulsarClient.newConsumer(Schema.BYTES)
 
 The messages in the retry letter topic contain some special properties that are automatically created by the client.
 
-Special property | Description
-:--------------------|:-----------
-`REAL_TOPIC` | The real topic name.
-`ORIGIN_MESSAGE_ID` | The origin message ID. It is crucial for message tracking.
-`RECONSUMETIMES`   | The number of retries to consume messages.
-`DELAY_TIME`      | Message retry interval in milliseconds.
+| Special property    | Description                                                |
```
Comment (asafm) [inline site2/docs/concepts-messaging.md:411]: image

File: site2/docs/concepts-messaging.md [hunk 6]
```diff
@@ -514,12 +514,12 @@ As in other pub-sub systems, topics in Pulsar are named channels for transmittin
...
-`tenant`             | The topic tenant within the instance. Tenants are essential to multi-tenancy in Pulsar, and spread across clusters.
-`namespace`          | The administrative unit of the topic, which acts as a grouping mechanism for related topics. Most topic configuration is performed at the [namespace](#namespaces) level. Each tenant has one or more namespaces.
-`topic`              | The final part of the name. Topic names have no special meaning in a Pulsar instance.
+| Topic name component            | Description                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                     |
```
Comment (asafm) [inline site2/docs/concepts-messaging.md:517]: image

File: site2/docs/concepts-messaging.md [hunk 7]
```diff
@@ -818,10 +818,10 @@ The subscription mode indicates the cursor type.
...
-`Durable` | The cursor is durable, which retains messages and persists the current position. <br />If a broker restarts from a failure, it can recover the cursor from the persistent storage (BookKeeper), so that messages can continue to be consumed from the last consumed position. | `Durable` is the **default** subscription mode.
-`NonDurable` | The cursor is non-durable. <br />Once a broker stops, the cursor is lost and can never be recovered, so that messages **can not** continue to be consumed from the last consumed position. | Reader’s subscription mode is `NonDurable` in nature and it does not prevent data in a topic from being deleted. Reader’s subscription mode **can not** be changed.
+| Subscription mode | Description                                                                                                                                                                                                                                                                   | Note                                                                                                                                                                |
```
Comment (asafm) [inline site2/docs/concepts-messaging.md:821]: image

File: site2/docs/concepts-messaging.md [hunk 8]
```diff
@@ -941,22 +941,22 @@ When publishing to partitioned topics, you must specify a *routing mode*. The ro
...
-`SinglePartition`     | If no key is provided, the producer will randomly pick one single partition and publish all the messages into that partition. While if a key is specified on the message, the partitioned producer will hash the key and assign message to a particular partition.
-`CustomPartition`     | Use custom message router implementation that will be called to determine the partition for a particular message. User can create a custom routing mode by using the [Java client](client-libraries-java.md) and implementing the {@inject: javadoc:MessageRouter:/client/org/apache/pulsar/client/api/MessageRouter} interface.
+| Mode                  | Description                                                                                                                                                                                                                                                                                                                                                                                                                                                       |
```
Comment (asafm) [inline site2/docs/concepts-messaging.md:944]: image

File: site2/docs/concepts-messaging.md [hunk 9]
```diff
@@ -941,22 +941,22 @@ When publishing to partitioned topics, you must specify a *routing mode*. The ro
...
 
 If there is a key attached to message, the messages will be routed to corresponding partitions based on the hashing scheme specified by {@inject: javadoc:HashingScheme:/client/org/apache/pulsar/client/api/HashingScheme} in {@inject: javadoc:ProducerBuilder:/client/org/apache/pulsar/client/api/ProducerBuilder}, when using either `SinglePartition` or `RoundRobinPartition` mode.
 
-Ordering guarantee | Description | Routing Mode and Key
-:------------------|:------------|:------------
-Per-key-partition  | All the messages with the same key will be in order and be placed in same partition. | Use either `SinglePartition` or `RoundRobinPartition` mode, and Key is provided by each message.
-Per-producer       | All the messages from the same producer will be in order. | Use `SinglePartition` mode, and no Key is provided for each message.
+| Ordering guarantee | Description                                                                          | Routing Mode and Key                                                                             |
```
Comment (asafm) [inline site2/docs/concepts-messaging.md:956]: image

**Review by tisonkun — APPROVED** [review]

**Review by momo-jun — APPROVED** [review]

**Pull request merged**